### PR TITLE
Extend `ibc::mock` types with more data fields

### LINKED ISSUE #186: Extend `ibc::mock` types with more data fields
We are going over some refactoring at `ibc-rs` to improve the ibc-testkit and existing tests.

During this, I realized that the current mock client types are too simple to test all IBC specifications. I want to extend the [current mock client types](https://github.com/cosmos/ibc-proto-rs/blob/main/definitions/mock/ibc.mock.proto) as follows.

```protobuf
message Header {
  ibc.core.client.v1.Height height = 1;
  google.protobuf.Timestamp timestamp = 2;
}

message ClientState {
  Header header = 1;
  google.protobuf.Duration trusing_period = 2;
  google.protobuf.Duration max_clock_drift = 3;
  bool frozen = 4;
}

message ConsensusState {
  Header header = 1;
  bytes consensus_bytes = 2;
}
```

I add `trusting_period` and `max_clock_drift` as `timestamp` is already part of the `Header`. `consensus_bytes` and `frozen` fields are the main fields - that I require. They are needed to test merkle proofs and frozen clients.

### PR DESCRIPTION
**Pull request by rnbguy**
Closes #186

### CODE REVIEW

**Comment by rnbguy** [comment]
I am a bit confused about how to run the build step. When I run `./scripts/sync-protobuf.sh`, I see changes at places other than `src/prost/ibc.mock*.rs` and `src/prost/proto_descriptor.bin` :sweat_smile: 

cc @romac @Farhad-Shabani

**Comment by Farhad-Shabani** [comment]
> I am a bit confused about how to run the build step. When I run `./scripts/sync-protobuf.sh`, I see changes at places other than `src/prost/ibc.mock*.rs` and `src/prost/proto_descriptor.bin` 😅
> 
> cc @romac @Farhad-Shabani

Make sure you are using the latest version of proto compiler (`libprotoc` v0.25)

**Comment by rnbguy** [comment]
I am using `v0.25.1` - do I have to get the latest `v0.25.2`?

File: definitions/mock/ibc.mock.proto [hunk 1]
```diff
@@ -1,19 +1,25 @@
...
 package ibc.mock;
 
+import "google/protobuf/duration.proto";
+import "google/protobuf/timestamp.proto";
 import "ibc/core/client/v1/client.proto";
 
 message Header {
   ibc.core.client.v1.Height height = 1;
-  uint64 timestamp = 2; 
+  google.protobuf.Timestamp timestamp = 2;
```
Comment (Farhad-Shabani) [inline definitions/mock/ibc.mock.proto]: Why is it necessary for the time to be of type `Timestamp` here?

File: definitions/mock/ibc.mock.proto [hunk 2]
```diff
@@ -1,19 +1,25 @@
...
 }
 
 message ClientState {
   Header header = 1;
+  google.protobuf.Duration trusing_period = 2;
+  google.protobuf.Duration max_clock_drift = 3;
```
Comment (Farhad-Shabani) [inline definitions/mock/ibc.mock.proto]: IMO, `trusting_period` and `max_clock_drift` is more of client specific fields, let’s keep the mock structure as general as possible and if some tests cannot be covered by mock, instead having them for tendermint `ClientState`.

File: definitions/mock/ibc.mock.proto [hunk 3]
```diff
@@ -1,19 +1,25 @@
...
+  bool frozen = 4;
 }
```
Comment (Farhad-Shabani) [inline definitions/mock/ibc.mock.proto:15]: Instead, I propose adding the `latest_height`, which is generic enough to be included, allowing us to track the latest height at which the client was updated.

File: definitions/mock/ibc.mock.proto [hunk 4]
```diff
@@ -1,19 +1,25 @@
...
 
 message ConsensusState {
   Header header = 1;
+  bytes bytes = 2;
```
Comment (Farhad-Shabani) [inline definitions/mock/ibc.mock.proto]: Renaming the `bytes` field to `root`? Given that consensus states usually track the state machine by this field. (Although I acknowledge that solomachine, for example, does not have this)

**Review by Farhad-Shabani — CHANGES_REQUESTED** [review]
Thank you @rnbguy for leveling up our mock definition. Left a few comments.

**Comment by Farhad-Shabani** [comment]
> I am using `v0.25.1` - do I have to get the latest `v0.25.2`?

Per our discussion, the issue of non-relevant changes should be resolved by placing `"google/*"` imports under the `"ibc/*"`. Perhaps we later should consider using `buf.yaml` for linting.

File: definitions/mock/ibc.mock.proto (hunk 1 above)
Comment (rnbguy) [reply definitions/mock/ibc.mock.proto]: there is no particular reason - except google protobuf types are standard.

File: definitions/mock/ibc.mock.proto (hunk 3 above)
Comment (rnbguy) [reply definitions/mock/ibc.mock.proto:15]: do you mean `latest_height` of the host? the height of the `ClientState` is already present inside `header` field.

File: definitions/mock/ibc.mock.proto (hunk 2 above)
Comment (rnbguy) [reply definitions/mock/ibc.mock.proto]: I wanted to model client expiration - that's why I wanted to add `trusting_period`.

I guess we can remove `max_clock_drift` but implying `max_clock_drift` is zero - as in client updates with future timestamp are invalid.

File: definitions/mock/ibc.mock.proto (hunk 4 above)
Comment (rnbguy) [reply definitions/mock/ibc.mock.proto]: hmm. ok. but, I am realizing we should specify the behavior of the Mock host and its light client. In that way, we can conclude what exact data we need to simulate it.

Comment (rnbguy) [reply definitions/mock/ibc.mock.proto]: and this leads to another question - should we even bother with a Mock host - when we can just have tm host ?

File: definitions/mock/ibc.mock.proto (hunk 3 above)
Comment (Farhad-Shabani) [reply definitions/mock/ibc.mock.proto:15]: Ah, you are right. we already have that 👍🏻

File: definitions/mock/ibc.mock.proto (hunk 1 above)
Comment (Farhad-Shabani) [reply definitions/mock/ibc.mock.proto]: Understood. Though still using the `Timestamp` can be a bit limiting IMO when trying to whip up quick tests. It's not exactly the most convenient. I think the mock setup should be as barebones as possible, with minimal dependencies and a straightforward structure.

**Comment by rnbguy** [comment]
@Farhad-Shabani as per our discussion, I simplified the definitions :slightly_smiling_face:

**Review by Farhad-Shabani — APPROVED** [review]
Thx @rnbguy. Looks good to me now.

**Comment by rnbguy** [comment]
@Farhad-Shabani thanks for the nits :sweat_smile: totally missed them ! I don't have access to this repo - please, go ahead with the merge. :handshake:

**Pull request merged**